internal/resource/config/configmap_builder.go

```diff
@@ -350,6 +350,25 @@ func (b *ConfigmapBuilder) Update(object client.Object) error {
 		}
 	}
 
+	if b.instance.Spec.PProf != nil {
+		var pprofConfig = config.PProf{}
+
+		if b.instance.Spec.PProf.Port > 0 {
+			pprofConfig.Port = b.instance.Spec.PProf.Port
+		}
+
+		if b.instance.Spec.PProf.Host != "" {
+			pprofConfig.Host = b.instance.Spec.PProf.Host
+		}
+
+		// Port is required if the object is defined.
+		// Host is optional.
+		// Do not set a pprof config if the port is 0.
+		if pprofConfig.Port > 0 {
```

Comment:
Is this condition mandatory ? Maybe this could be moved to the first condition.
WDYT ?

Reply:
Happy to remove it and reorder the conditionals.